### PR TITLE
fix: phishing test to not check c2 domains

### PR DESCRIPTION
**Pull request by AugmentedMode**
## **Description**

This PR modifies how phishing detection and blocklist checks are handled within the phishingController. Specifically, it ensures that only the isBlockedRequest function checks the blocklist against network requests. The test function will no longer handle for network requests but more specifically only be ran on `main_fame` and `sub_frame` requests. This allowed SEAL to submit C2 domains, however going forward they will need to submit C2 domains directly to us to be ingested. 

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27846?quickstart=1)

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Review by 0xOhm — APPROVED** [review]

File: app/scripts/background.js [hunk 1]
```diff
@@ -266,13 +266,17 @@ function maybeDetectPhishing(theController) {
...
-      const phishingTestResponse = theController.phishingController.test(
-        details.url,
-      );
 
       const blockedRequestResponse =
         theController.phishingController.isBlockedRequest(details.url);
 
+      let phishingTestResponse;
+      if (details.type === 'main_frame' || details.type === 'sub_frame') {
+        phishingTestResponse = theController.phishingController.test(
```
Comment (mindofmar) [inline app/scripts/background.js:275]: I think you have this backwards??

`isBlockedRequest` is for C2s, but `test` is for the EPD list.

Comment (AugmentedMode) [reply app/scripts/background.js:275]: Right, but this allowed us to add C2s to EPD which would enable SEAL, however we want SEAL to have to directly send us the C2s, so we dont want to check the EPD list against network requests which is what is happening.

Comment (mindofmar) [reply app/scripts/background.js:275]: C2 requests can still come from `main_frame` and `sub_frame` though, so how does this solve that?

Comment (AugmentedMode) [reply app/scripts/background.js:275]: network requests only come from `xmlhttprequest` which we do not want EPD to check against, however we want EPD to check again `main_frame` and `sub_frame`

Comment (AugmentedMode) [reply app/scripts/background.js:275]: It's okay if EPD blocks a C2 from a main_frame or sub_frame.

**Review by mindofmar — APPROVED** [review]

**Review by cryptotavares — APPROVED** [review]

**Pull request merged**